### PR TITLE
Add Maldivian Holidays

### PR DESCRIPTION
**Pull request by ahmedmansoor**
- [x] New Year's Day
- [x] First day of Ramadan
- [x] Last 10 Days of Ramadan - Day 1
- [x] Last 10 Days of Ramadan - Day 2
- [x] Last 10 Days of Ramadan - Day 3
- [x] Last 10 Days of Ramadan - Day 4
- [x] Last 10 Days of Ramadan - Day 5
- [x] Last 10 Days of Ramadan - Day 6
- [x] Last 10 Days of Ramadan - Day 7
- [x] Last 10 Days of Ramadan - Day 8
- [x] Last 10 Days of Ramadan - Day 9
- [x] Last 10 Days of Ramadan - Day 10
- [x] Eid al-Fitr
- [x] On the occasion of Eid al-Fitr - 2nd day
- [x] On the occasion of Eid al-Fitr - 3rd day
- [x] Labour Day
- [x] Hajj Day
- [x] Eid al-Adha
- [x] On the occasion of Eid al-Adha - 2nd day
- [x] On the occasion of Eid al-Adha - 3rd day
- [x] Islamic New Year
- [x] Independence Day
- [x] On the occasion of the Independence Day
- [x] National Day
- [x] Prophet Muhammad (SAW)'s Birthday
- [x] The Day Maldives Embraced Islam
- [x] Victory Day
- [x] Republic Day

reference: 
[Public Holidays in the Maldives](https://en.wikipedia.org/wiki/Public_holidays_in_the_Maldives)
[Academic Calendar | Min of Education](https://www.moe.gov.mv/en/category/view/13)

### CODE REVIEW

**Comment by ahmedmansoor** [comment]
snapshot

**Comment by thecaliskan** [comment]
Hi @ahmedmansoor 

- It does not work for the year 2000. Because there are two Ramadan holidays
- Package "geniusts/hijri-dates" gives wrong result for 1 Sewwal in 2025 and some years.

File: src/Countries/Maldives.php [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+class Maldives extends Country
+{
+    protected $adjustmentDays = 1; // Adjustments based on moon sighting
+
+    public function countryCode(): string
+    {
+        return 'mv';
+    }
+
+    /** @return array<string, string|CarbonImmutable> */
```
Comment (Nielsvanpach) [inline src/Countries/Maldives.php:17]: annotations can be removed

File: src/Countries/Maldives.php [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    protected function allHolidays(int $year): array
+    {
+        // Fixed-date holidays
```
Comment (Nielsvanpach) [inline src/Countries/Maldives.php:20]: annotation can be removed

File: src/Countries/Maldives.php [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+            'On the occasion of the Independence Day' => '07-27',
+            'Victory Day' => '11-03',
+            'Republic Day' => '11-11',
+        ], $this->variableHolidays($year));
+    }
+
+    /** @return array<string, CarbonImmutable> */
+    protected function variableHolidays(int $year): array
+    {
+        // Ramadan
```
Comment (Nielsvanpach) [inline src/Countries/Maldives.php:34]: Hi there. Could you write a test to generate the dates for 2023, 2024 and 2025 and verify if the generated results in the snapshots are correct? We don't have a proper solution for Islamic holidays yet, so I'm curious if this works.

**Review by Nielsvanpach — COMMENTED** [review]
Thanks! Could you get your fork in sync with this repo and rebase on our main branch? This way we can run the tests here

File: src/Countries/Maldives.php (hunk 3 above)
Comment (ahmedmansoor) [reply src/Countries/Maldives.php]: Hi there. 

The updated version results align with published public holidays ([Academic Calendar | Min of Education](https://www.moe.gov.mv/en/category/view/13)) of all years tested (2021, 2022, 2023, 2024, and 2025).

File: src/Countries/Maldives.php (hunk 2 above)
Comment (ahmedmansoor) [reply src/Countries/Maldives.php:19]: done.

File: src/Countries/Maldives.php (hunk 1 above)
Comment (ahmedmansoor) [reply src/Countries/Maldives.php]: done

**Comment by ahmedmansoor** [comment]
> Hi @ahmedmansoor
> 
> * It does not work for the year 2000. Because there are two Ramadan holidays
> * Package "geniusts/hijri-dates" gives wrong result for 1 Sewwal in 2025 and some years.

@thecaliskan I've tested for 2000 in the latest version, you may check. Also, if we need adjustments for the second half of 2025's hijri calendar, it will be announced after 6 months this year or at the end of this year. Current results for 2025 align with the dates published at this time.

**Comment by Nielsvanpach** [comment]
A few other PRs (https://github.com/spatie/holidays/pull/56) are working on Islamic holidays too. It might be useful to work together on this one. I would prefer a single implementation for those calculations which all countries with Islamic holidays could reuse.

**Comment by ahmedmansoor** [comment]
> A few other PRs (#56) are working on Islamic holidays too. It might be useful to work together on this one. I would prefer a single implementation for those calculations which all countries with Islamic holidays could reuse.

@Nielsvanpach OK, I will be checking on that. 

It's doable hopefully, although, it's common for countries to have a day or two difference in Islamic holiday dates because they are based on moon sightings based on regions.

**Comment by ahmedmansoor** [comment]
> A few other PRs (#56) are working on Islamic holidays too. It might be useful to work together on this one. I would prefer a single implementation for those calculations which all countries with Islamic holidays could reuse.

@Nielsvanpach it's doable, hopefully, though Islamic holiday dates may vary by a day or two among countries due to regional differences in moon sightings.

**Comment by ahmedmansoor** [comment]
> A few other PRs (#56) are working on Islamic holidays too. It might be useful to work together on this one. I would prefer a single implementation for those calculations which all countries with Islamic holidays could reuse.

@Nielsvanpach OK I'll check on that. It's doable, hopefully, though Islamic holiday dates may vary by a day or two among countries due to regional differences in moon sightings.

**Comment by ahmedmansoor** [comment]
Hi @Nielsvanpach. I've created a new PR #186, as I messed up this one during git rebasing. Sorry about that.
I've also created a Hijri Calendar trait so others can use [geniusts/hijri-dates](https://github.com/GeniusTS/hijri-dates) to calculate Hijri dates.

**Pull request closed without merging**